### ISSUE TITLE
[Bug]: test_search_binary_with_pagination not stable

### ISSUE THREAD

**Issue by longjiquan** [milvus-io/milvus#22677]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-22639/4/pipeline

### Expected Behavior

Below assertion is enough.
```python
assert sorted(search_res[0].distances, key=numpy.float32) == sorted(res[0].distances[offset:], key=numpy.float32)
```

**Comment by longjiquan** [comment]
For `jaccard` metric type, the similarity is negatively related to the distance. Given a vector to be searched, the distances of returned result should be sorted by ascending order. In this case, if there are many vectors whose intersection with the searched vector is zero, then they will be returned later in the result set, so `unique_a` and `unique_b` will contain different id.

**Comment by yanliang567** [comment]
/assign @NicoYuan1986 
please help tp update the tests accoridnlgy.
/unassign

**Comment by NicoYuan1986** [comment]
The issue should be fixed after pr #22681 merged
